### PR TITLE
Allow output to specific file path with CLI

### LINKED ISSUE #1672: Request: --out CLI option should allow filename control
### The Problem
Using the CLI `--out` option, it is currently not possible to control the filename of the compiled CSS file. All we can do is set the folder where the compiled CSS file is placed.

Thus, `basic.styl` always becomes `basic.css`. 

Many times, people want a different output filename. For example: `basic.min.css` or similar names that match patterns expected by various Content Management Systems. (Or to follow their team's file-naming conventions.)

### The Current Workaround
I develop CodeKit and, until now, I've worked around this Stylus limitation by having the compiler compile to a temporary folder. I then move the file into the correct spot and rename it as appropriate.

### Why The Workaround No Longer Works
With the new source map capabilities, the approach above won't work. The compiler will calculate the wrong paths because I haven't told it the real, final output location. I can't simply move and rename output files any longer.

I've actually hacked a new workaround. I still use the "compile and relocate" approach, but I have CodeKit open the CSS file and the source map and edit their contents as appropriate. This works, but it's slower than having the Stylus CLI do this on its own (because it involves 2x the disk-write operations). 

### Why This Matters
Stylus is the only language that has this limitation. Both Less and Sass provide users a way to customize the output filename. If Stylus is going to remain competitive, I think it needs to offer this flexibility.

### PR DESCRIPTION
**Pull request by TheAlphaNerd**
Closes #1672

### CODE REVIEW

**Comment by TheAlphaNerd** [comment]
This should only work when an out is specified as a file, and there is only a single input.  It should not effect current functionality.  Unfortunately since there are no tests yet there is no way to prove it, although I did do some manual regression testing with a local repo and some build steps

File: bin/stylus [hunk 1]
```diff
@@ -693,11 +700,7 @@ function writeFile(file, css) {
 */
 
 function writeSourcemap(file, sourcemap) {
-  // --out support
-  var out = basename(file, extname(file)) + '.css.map'
-    , path = dest
-      ? join(dest, out)
-      : join(dirname(file), out);
+  var path = createPath(file);
```
Comment (Panya) [inline bin/stylus]: There should be `<file>.css.map` not `<file>.css` so we should introduce an optional argument with extname to `createPath`.

**Comment by TheAlphaNerd** [comment]
I think everything checks out now.  It was manually tested again with a single file and .map outputs as expected.

Nice catch! (Over Abstracted)

**Comment by TheAlphaNerd** [comment]
Hold off on merging this... I'm tweaking it a bit to try and kill 3 lines

**Comment by TheAlphaNerd** [comment]
Ready to merge

**Comment by TheAlphaNerd** [comment]
Actually hold on that, I think there is a problem in the logic that will break source maps for multi files

**Comment by TheAlphaNerd** [comment]
This appears to be working as expected now.  With both single files and multiple files (manually tested)

**Comment by bdkjones** [comment]
It looks like the help message for the CLI did not get updated to reflect this new functionality. Should do that for the next release.

**Pull request merged**